scripts/gradle/gradle-check.sh

```diff
@@ -20,6 +20,39 @@ TRIGGER_TOKEN=$1
 PR_TITLE_NEW=`echo $pr_title | tr -dc '[:alnum:] ' | tr '[:upper:]' '[:lower:]'`
 PAYLOAD_JSON="{\"pr_from_sha\": \"$pr_from_sha\", \"pr_from_clone_url\": \"$pr_from_clone_url\", \"pr_to_clone_url\": \"$pr_to_clone_url\", \"pr_title\": \"$PR_TITLE_NEW\", \"pr_number\": \"$pr_number\"}"
 
+perform_curl_and_process_with_jq() {
+    local url=$1
+    local jq_filter=$2
+    local max_retries=$3
+    local count=0
+    local success=false
+
+    while [ $count -lt $max_retries ]; do
+        response=$(curl -s -XGET "${url}api/json")
+        curl_exit_code=$?
+
+        if [ $curl_exit_code -eq 0 ]; then
+            processed_response=$(echo "$response" | jq --raw-output "$jq_filter")
+            jq_exit_code=$?
+
+            if [ $jq_exit_code -eq 0 ]; then
+                success=true
+                echo "$processed_response"
+                break
+            fi
+        fi
+
+        echo "Attempt $((count+1))/$max_retries failed. Curl exit code: $curl_exit_code, jq exit code: $jq_exit_code. Retrying..."
+        count=$((count+1))
+        sleep 5
+    done
+
+    if [ "$success" != true ]; then
+        echo "Failed to retrieve and process data after $max_retries attempts."
+        return 1
```

Comment:
To follow up on previous comment, we can just do a `exit 1` here if `$success` is not true value.

Reply:
yes, we can just exit from here .. let me do that ..